test/extended/util/oauthserver/helpers.go

```diff
@@ -25,14 +28,37 @@ func init() {
 	utilruntime.Must(osinv1.Install(osinScheme))
 }
 
-func RequestTokenForUser(reqOpts *tokencmd.RequestTokenOptions, username, password string) (string, error) {
+func RequestTokenForUser(reqOpts *tokencmd.RequestTokenOptions, username, password string, retryTimeout time.Duration) (string, error) {
 	reqOpts.Handler = &tokencmd.BasicChallengeHandler{
 		Host:     reqOpts.ClientConfig.Host,
 		Username: username,
 		Password: password,
 	}
-
-	return reqOpts.RequestToken()
+	var (
+		token string
+		err   error
+	)
+	waitFn := func() (bool, error) {
+		e2e.Logf("Waiting up to %v for oauth token", retryTimeout)
+		token, err = reqOpts.RequestToken()
```

Comment:
this wont' work, you're using a stateful thing statelessly.

Reply:
Meaning the way we were passing around reqOpts is wrong? Or the use of RequestToken() is wrong here?  I've moved reqOpts to within the RequestTokenForUser fn, above.  I may need clarification from you or @enj to get this right : (

Reply:
What is actually flaking? `RequestToken` does a lot of things. We have to know better than putting a huge out loop here

Reply:
[@sanchezl 's PR](https://github.com/openshift/origin/pull/23528) has likely fixed the flake, this PR is now only to remove 'Flaky' tag : )